spring-cloud-kubernetes-commons/src/test/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesDiscoveryPropertiesMetadataTests.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.commons.discovery;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Configuration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.cloud.kubernetes.commons.discovery.KubernetesDiscoveryProperties.Metadata;
+
+/**
+ * @author wind57
+ */
+class KubernetesDiscoveryPropertiesMetadataTests {
+
+	@Test
+	void testDefaultConstructor() {
+		Metadata m = new Metadata();
+		assertThat(m.addLabels()).isTrue();
+		assertThat(m.labelsPrefix()).isNull();
+		assertThat(m.addAnnotations()).isTrue();
+		assertThat(m.annotationsPrefix()).isNull();
+		assertThat(m.addPorts()).isTrue();
+		assertThat(m.portsPrefix()).isEqualTo("port.");
+	}
+
+	@Test
+	void testSpringBindingFields() {
+		new ApplicationContextRunner().withUserConfiguration(Config.class)
```

Comment:
proves that even if I set a single property, all defaults otherwise are preserved.